market_nextdoor_api/models.py

```diff
@@ -88,4 +88,59 @@ class Preorder2Item(models.Model):
     updated_at = models.DateTimeField(auto_now=True)
 
     def __str__(self):
-        return f"{self.quantity_requested} of {self.item.item_name} in {self.preorder}"
\ No newline at end of file
+        return f"{self.quantity_requested} of {self.item.item_name} in {self.preorder}"
+
+class Customer2(models.Model):
+    first_name = models.CharField(max_length=50)
+    last_name = models.CharField(max_length=50)
+    phone = models.CharField(max_length = 10, null=True)
+    email = models.CharField(max_length=50)
+    password = models.CharField(max_length=50, null=False)
+    location = models.CharField(max_length=255, null=True)
+    date_created = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.first_name + " " + self.last_name
+    
+class Vendor2(models.Model):
+    vendor_name = models.CharField(max_length=50)
+    first_name = models.CharField(max_length=50)
+    last_name = models.CharField(max_length=50)
+    phone = models.CharField(max_length=12, null=True)
+    email = models.CharField(max_length=50)
+    password = models.CharField(max_length=50, null=False)
+    location = models.CharField(max_length=255, null=True)
+    date_created = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.vendor_name
+    
+class Market2(models.Model):
+    market_name = models.CharField(max_length=50)
+    location = models.CharField(max_length =100)
+    details = models.CharField(max_length=100)
+    start_date = models.DateField(null=True)
+    end_date = models.DateField(null=True)
+    vendors = models.ManyToManyField(Vendor2, through='Vendor2Market')
+    customers = models.ManyToManyField(Customer2, through='Customer2Market')
+    date_created = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.market_name
+    
+class Customer2Market(models.Model):
+    customer = models.ForeignKey(Customer2, on_delete=models.CASCADE, null=False)
+    market = models.ForeignKey(Market2, on_delete=models.CASCADE, null=False)
+
+    def __str__(self):
+        return f"{self.customer.first_name} {self.customer.last_name} at {self.market.market_name}"
+
+class Vendor2Market(models.Model):
```

Comment:
Thinking of other field we might want for these joins tables such as the quantity in the pre_order_items table, but nothing coming to mind right off the top of my head.